src/api/query.rs

```diff
@@ -96,117 +102,146 @@ impl QueryHelper {
                     } => foundation_transaction_rewards,
                 }
             }
-            Address::PoolAccrueAccount(baker_id) => {
-                match self.client.clone().get_pool_status(baker_id, &block_hash).await {
-                    Ok(i) => match i {
-                        PoolStatus::BakerPool {
-                            status,
-                        } => match status.current_payday_status {
-                            None => Amount::from_ccd(0),
-                            Some(s) => s.transaction_fees_earned,
-                        },
-                        PoolStatus::PassiveDelegation {
-                            status,
-                        } => status.current_payday_transaction_fees_earned,
+            Address::PoolAccrueAccount(baker_id) => match baker_id {
+                Some(id) => match self.client.clone().get_pool_info(&block_hash, id).await {
+                    Ok(i) => match i.response.current_payday_status {
+                        None => Amount::from_ccd(0),
+                        Some(s) => s.transaction_fees_earned,
                     },
                     Err(err) => match err {
                         QueryError::RPCError(err) => return Err(err.into()),
                         QueryError::NotFound => Amount::from_micro_ccd(0),
                     },
-                }
-            }
+                },
+                None => match self.client.clone().get_passive_delegation_info(&block_hash).await {
+                    Ok(i) => i.response.current_payday_transaction_fees_earned,
+                    Err(err) => match err {
+                        QueryError::RPCError(err) => return Err(err.into()),
+                        QueryError::NotFound => Amount::from_micro_ccd(0),
+                    },
+                },
+            },
         };
         Ok((block_info, amount))
     }
 
+    pub async fn query_consensus_info(&self) -> ApiResult<ConsensusInfo> {
+        map_query_result(
+            self.client.clone().get_consensus_info().await,
+            ApiError::UnexpectedSdkError,
```

Comment:
It does return a `ClientRPCError` _unless_ the `get_consensus_info` call returns a `NotFound`. Since it should not be possible for it to return a `NotFound`, this error shouldn't happen ever in practice. I wasn't entirely sure what to do here.

Reply:
We should not annoy the clients with this internal detail.

We should map this to some internal server error and log an error. Isn't there already an internal error?